src/src/streamapi/hw/hillel/Main.java

```diff
@@ -0,0 +1,60 @@
+package streamapi.hw.hillel;
+
+import java.time.LocalDate;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Main {
+    public static void main(String[] args) throws Exception {
+        List<Product> pr = new ArrayList<>();
+
+        pr.add(new Product(1, "Book", 25, false, LocalDate.now()));
+        pr.add(new Product(2, "Movie", 10, true, LocalDate.now()));
+        pr.add(new Product(3, "Book", 500, true, LocalDate.now().minusDays(50)));
+        pr.add(new Product(4, "Music", 5, false, LocalDate.now().minusMonths(6)));
+        pr.add(new Product(5, "Book", 70, false, LocalDate.now().minusDays(10)));
+
+        System.out.println(getCheapestBook(pr));
+        System.out.println(groupProd(pr));
+        System.out.println(calculateTotalPrice(pr));
+        System.out.println(getExpensiveBooks(pr));
+        System.out.println(getLastThreeAddedProducts(pr));
+    }
+
+    public static List<Product> getExpensiveBooks(List<Product> products) {
+        return products.stream()
+                .filter(product -> product.getType().equals("Book") && product.getPrice() > 250)
+                .collect(Collectors.toList());
+    }
+
+    public static List<Product> getLastThreeAddedProducts(List<Product> products) {
+        return products.stream()
+                .sorted(Comparator.comparing(Product::getAddedDate).reversed())
```

Comment:
Sorry, but you should not sort the elements in the list because you lost the initial order, so remove `sorted()`